File: javascript/space-age/space-age.js
const EARTH_ORBITAL_S = 31557600;
const MERCURY_ORBITAL_S = 7600543.81992;
const VENUS_ORBITAL_S = 19414149.052176;
const MARS_ORBITAL_S = 59354032.69008;
const JUPITER_ORBITAL_S = 374355659.124;
const SATURN_ORBITAL_S = 929292362.8848;
const URANUS_ORBITAL_S = 2651370019.3296;
const NEPTUNE_ORBITAL_S = 5200418560.032;

function SpaceAge(seconds) {
  this.seconds = seconds;
}

SpaceAge.prototype.onEarth = function onEarth() {
  return this.calcAge(EARTH_ORBITAL_S);
};

SpaceAge.prototype.onMercury = function onMercury() {
  return this.calcAge(MERCURY_ORBITAL_S);
};

SpaceAge.prototype.onVenus = function onVenus() {
  return this.calcAge(VENUS_ORBITAL_S);
};

SpaceAge.prototype.onMars = function onMars() {
  return this.calcAge(MARS_ORBITAL_S);
};

SpaceAge.prototype.onJupiter = function onJupiter() {
  return this.calcAge(JUPITER_ORBITAL_S);
};

SpaceAge.prototype.onSaturn = function onSaturn() {
  return this.calcAge(SATURN_ORBITAL_S);
};

SpaceAge.prototype.onUranus = function onUranus() {
  return this.calcAge(URANUS_ORBITAL_S);
};

SpaceAge.prototype.onNeptune = function onNeptune() {
  return this.calcAge(NEPTUNE_ORBITAL_S);
};

SpaceAge.prototype.calcAge = function calcAge(orbital) {
  return parseFloat((this.seconds / orbital).toFixed(2));
};

module.exports = SpaceAge;

File: csharp/space-age/SpaceAge.cs
using System;

public class SpaceAge
{
    private readonly double _earthOrbital = 31557600;
    private readonly double _mercuryOrbital = 7600543.81992;
    private readonly double _venusOrbital = 19414149.052176;
    private readonly double _marsOrbital = 59354032.69008;
    private readonly double _jupiterOrbital = 374355659.124;
    private readonly double _saturnOrbital = 929292362.8848;
    private readonly double _uranusOrbital = 2651370019.3296;
    private readonly double _neptuneOrbital = 5200418560.032;
    private long _seconds;
    public SpaceAge(long seconds)
    {
        _seconds = seconds;
    }
    public double OnEarth() => this.CalcAge(_earthOrbital);
    public double OnMercury() => this.CalcAge(_mercuryOrbital);
    public double OnVenus() => this.CalcAge(_venusOrbital);
    public double OnMars() => this.CalcAge(_marsOrbital);
    public double OnJupiter() => this.CalcAge(_jupiterOrbital);
    public double OnSaturn() => this.CalcAge(_saturnOrbital);
    public double OnUranus() => this.CalcAge(_uranusOrbital);
    public double OnNeptune() => this.CalcAge(_neptuneOrbital);
    private double CalcAge(double orbital) => Math.Round(_seconds / orbital, 2);
}